themes/default/content/docs/concepts/how-pulumi-works.md

```diff
@@ -17,35 +17,37 @@ aliases:
 - /docs/intro/concepts/how-pulumi-works/
 ---
 
-Pulumi uses a desired state model for managing infrastructure. A Pulumi program is executed by a _language host_ to compute a desired state for a stack's infrastructure. The _deployment engine_ compares this desired state with the stack's current state and determines what resources need to be created, updated or deleted. The engine uses a set of _resource providers_ (such as [AWS](/docs/clouds/aws/get-started/), [Azure](/docs/clouds/azure/get-started/), [Kubernetes](/docs/clouds/kubernetes/get-started/), and so on) in order to manage the individual resources.  As it operates, the engine updates the _state_ of your infrastructure with information about all resources that have been provisioned as well as any pending operations.
+Pulumi uses a desired state `declarative` model for orchestrating and managing infrastructure with the flexibility to author your infrastructure code using programming languages you know such as TypeScript, Javascript, Python, Go, C#, and Java. This model provides the advantages of programming structures like loops, conditionals and functions as well as your IDE with autocomplete, type checking, and documentation. When you author a Pulumi program the end result will be the state you declare, regardless of the current state of your infrastructure. 
+
+A Pulumi program is executed by a _language host_ to compute a desired state for a stack's infrastructure. The _deployment engine_ compares this desired state with the stack's current state and determines what resources need to be created, updated or deleted. The engine uses a set of _resource providers_ (such as [AWS](/docs/clouds/aws/get-started/), [Azure](/docs/clouds/azure/get-started/), [Kubernetes](/docs/clouds/kubernetes/get-started/), and [+150 more](/registry/)) in order to manage the individual resources.  As it operates, the engine updates the _state_ of your infrastructure with information about all resources that have been provisioned as well as any pending operations.
 
 The following diagram illustrates the interaction between these parts of the system:
 
-<img src="/images/docs/reference/engine-block-diagram.png" alt="Pulumi engine and providers" width="600">
+<img src="/images/docs/reference/engine-block-diagram.png" alt="Pulumi IaC system architecture, the Pulumi engine and providers" width="600">
 
 In the next section, we will describe each of these components and see how they all fit together during an invocation of `pulumi up`.
 
-## Language Hosts
+## Language hosts
```

Comment:
The second word is made lowercase here, but not in other titles further down like `Resource Providers`. Is this an inconsistency now, or is there still a reason behind it? I'm not an native English speaker, so I might not know.